utils/preferences/tests/dtf/preferences.rs

```diff
@@ -0,0 +1,37 @@
+use icu_locid::{extensions::unicode, extensions_unicode_value};
+
+#[derive(Clone, Copy)]
+pub enum Calendar {
+    Gregory,
+    Buddhist,
+}
+
+impl TryFrom<&unicode::Value> for Calendar {
+    type Error = ();
+
+    fn try_from(i: &unicode::Value) -> Result<Self, Self::Error> {
+        match i {
+            _ if *i == extensions_unicode_value!("gregory") => Ok(Self::Gregory),
+            _ if *i == extensions_unicode_value!("buddhist") => Ok(Self::Buddhist),
+            _ => Err(()),
+        }
+    }
+}
+
+#[derive(Clone, Copy)]
+pub enum NumberingSystem {
```

Comment:
Nit: `NumberingSystem` never needs to be reified in code, except for potentially toggling between DecimalFormat and RuleBasedNumberFormat (which we don't support yet). Generally, it is an unbounded set that grows over time, so it should not be an enum. How can you express this in your model?

Reply:
How is it different from HourCycle?

Reply:
There are only 4 hour cycles, but there is an unbounded and constantly growing set of numbering systems (more like scripts).

Reply:
I think we have different mental models. The number of hour cycles is also arbitrary and can grow. I doubt it will, but it might.
Similarly, the number of numbering systems is rarely changing, and if it does we can add a new variant so my preference would be to keep it as an enum and have it be non-exhaustive.

But the question you raise is valid, so I changed it it to tinystr for the purpose of the example.

Reply:
About a year ago we agreed upon guidelines for "enums or identifiers"

https://github.com/unicode-org/icu4x/blob/main/docs/design/enums_or_ids.md

Reply:
I recall that. I don't think that guideline replaces architectural decision on what HourCycle is and what NumberingSystem is.